File: _includes/sort-benchmark/algorithms/comb.rs
fn comb_sort(a: &mut [usize]) {
    let mut gap = a.len();
    let mut swapped = true;
    while gap > 1 || swapped {
        gap = (gap * 10 / 13).max(1);
        swapped = false;
        for i in 0..a.len().saturating_sub(gap) {
            if a[i] > a[i + gap] {
                a.swap(i, i + gap);
                swapped = true;
            }
        }
    }
}

File: _includes/sort-benchmark/algorithms/comb.swift
func comb_sort(_ a: inout [Int]) {
    a.withUnsafeMutableBufferPointer { comb_sort($0) }
}

func comb_sort(_ a: UnsafeMutableBufferPointer<Int>) {
    var gap = a.count
    var swapped = true
    while gap > 1 || swapped {
        gap = max(gap * 10 / 13, 1)
        swapped = false
        let limit = max(a.count - gap, 0)
        for i in 0..<limit {
            if a[i] > a[i + gap] {
                a.swapAt(i, i + gap)
                swapped = true
            }
        }
    }
}
